integration-test.sh

```diff
@@ -233,6 +257,51 @@ function test_distributions {
     done
 }
 
+function test_distribution_fails_with_wrong_track_params {
+    local cfg
+    local distribution
+    # TODO check if randomization of track is possible
+    local track="geonames" # fixed value for now, as the available track params vary between tracks
+    local track_params
+    local defined_track_params
+    local undefined_track_params
+
+    random_configuration cfg
+    random_distribution dist
+
+    undefined_track_params="number_of-replicas:0" # - simulates a typo
+
+    if [[ ${track} == "geonames" ]]; then
+        defined_track_params="conflict_probability:45,"
+    fi
+
+    local track_params="${defined_track_params}${undefined_track_params}"
+    readonly err_msg="Rally didn't fail trying to use the undefined track-param ${undefined_track_params}. Check ${RALLY_LOG}."
+
+    info "test distribution [--configuration-name=${cfg}], [--distribution-version=${dist}], [--track=${track}], [--track-params=${track_params}], [--car=4gheap]"
+    kill_rally_processes
+
+    backup_rally_log
+    set +e
+    esrally --configuration-name="${cfg}" --on-error=abort --distribution-version="${dist}" --track="${track}" --track-params="${track_params}" --test-mode --car=4gheap
+    ret_code=$?
+    set -e
+
+    # we expect Rally to fail, with full details in its log file
+    if [[ ${ret_code} -eq 0 ]]; then
+        error "Rally didn't fail trying to use the undefined track-param ${undefined_track_params}. Check ${RALLY_LOG}."
+        error ${err_msg}
+        exit ${ret_code}
+    elif [[ ${ret_code} -ne 0 ]]; then
+        # need to use grep -P which is unavailable with macOS grep
+        if ! docker run --rm -v ${RALLY_LOG}:/rally.log:ro ubuntu grep -Pzoq '.*ERROR Some of your track parameter\(s\) "number_of-replicas" are not used by this track; perhaps you intend to use "number_of_replicas" instead\.\n\nAll track parameters you provided are:\n- conflict_probability\n- number_of-replicas\n\nAll parameters exposed by this track:\n- bulk_indexing_clients\n- bulk_size\n- cluster_health\n- conflict_probability\n- index_settings\n- ingest_percentage\n- number_of_replicas\n- number_of_shards\n- on_conflict\n- recency\n- source_enabled\n*' /rally.log; then
```

Comment:
I think we need to add a check whether Docker is available / running similarly to the proxy test? Also, I wonder whether it would still be more portable to use Perl for this? (we already use it in `set_up_metrics_store`).

Reply:
Addressed in https://github.com/elastic/rally/pull/688/commits/49c7994c7a380d61c4eaeaf68f1ad60a7c5d572d

Kept Docker for now but pinned the version of the image.